proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/verifyserviceprovider/Attributes.java

```diff
@@ -53,40 +55,82 @@ public Attributes(
-        return firstNames != null ? firstNames : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<String> getFirstNames() {
+        return new AttributesList(firstNames, "firstName");
     }
 
-    public List<Attribute<String>> getMiddleNames() {
-        return middleNames != null ? middleNames : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<String> getMiddleNames() {
+        return new AttributesList(middleNames, "middleName");
     }
 
-    public List<Attribute<String>> getSurnames() {
-        return surnames != null ? surnames : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<String> getSurnames() {
+        return new AttributesList(surnames, "surname");
     }
 
-    public List<Attribute<DateTime>> getDatesOfBirth() {
-        return datesOfBirth != null ? datesOfBirth : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<DateTime> getDatesOfBirth() {
+        return new AttributesList(datesOfBirth, "dateOfBirth");
     }
 
+    @JsonIgnore
     public Attribute<String> getGender() {
         return gender;
     }
 
-    public List<Attribute<Address>> getAddresses() {
-        return addresses;
-    }
-
-    public static String combineAttributeValues(List<Attribute<String>> attributes) {
-        return attributes.stream()
-                .filter(Objects::nonNull)
-                .map(Attribute::getValue)
-                .filter(s -> !s.isEmpty())
-                .collect(Collectors.joining(" "));
+    @JsonIgnore
+    public AttributesList<Address> getAddresses() {
+        return new AttributesList(addresses, "address");
     }
 
     // Prints date in EIDAS format YYYY-MM-dd
-    public static String getFormattedDate(DateTime date) {
+    public static String getDateInEidasFormat(DateTime date) {
         return ISODateTimeFormat.date().withChronology(ISOChronology.getInstanceUTC()).print(date);
     }
+
+    public class AttributesList<T> {
+
+        private final List<Attribute<T>> attributes;
+        private final String type;
+
+        AttributesList(List<Attribute<T>> attributes, String type) {
+            this.attributes = Optional.ofNullable(attributes).orElse(Collections.emptyList());
+            this.type = type;
+        }
+
+        public String getType() {
+            return type;
+        }
+
+        public List<Attribute<T>> getValidAttributes() {
+
+            var current = getAllAttributes().stream()
+                    .filter(Attribute::isCurrent)
+                    .collect(Collectors.toList());
+
+            var verifiedAndCurrent = current.stream()
+                    .filter(Attribute::isVerified)
+                    .collect(Collectors.toList());
+
+            if (verifiedAndCurrent.isEmpty()) {
+                ProxyNodeLogger.info("No verified and current attributes: " + createAttributesMessage());
+                return current;
+            } else {
+                return verifiedAndCurrent;
+            }
```

Comment:
No need for the `else` part here since the `if` part ends in `return`

Reply:
Updated after our review together, thanks!